usecase/pokemon_service.go

```diff
@@ -0,0 +1,36 @@
+package usecase
+
+import (
+	"github.com/nibble-4bits/ondemand-go-bootcamp/entity"
+)
+
+type pokemonService struct {
+	repo PokemonRepository
+}
+
+type PokemonService interface {
+	GetByID(id int) (entity.Pokemon, error)
+	GetAll() ([]entity.Pokemon, error)
+}
+
+func NewPokemonService(r PokemonRepository) pokemonService {
+	return pokemonService{repo: r}
+}
+
+func (s pokemonService) GetByID(id int) (entity.Pokemon, error) {
+	pokemonFound, err := s.repo.GetByID(id)
+	if err != nil {
+		return entity.Pokemon{}, err
```

Comment:
Errors returned from external packages should be wrapped. Please give it a look to `fmt.Errorf()` method and the `%w` printing verb.

Same for the `GetAll()` method a few lines next.